src/adjust.js

```diff
@@ -0,0 +1,30 @@
+var _curry3 = require('./internal/_curry3');
+
+/**
+ * Applies a function to the value at the given index of an array,
+ * returning a new copy of the array with the element at the given
+ * index replaced with the result of the function application.
+ *
+ * @func
+ * @memberOf R
+ * @category List
+ * @sig (a -> a) -> Int -> [a] -> [a]
+ * @param {Function} fn The function to apply.
+ * @param {Number} idx The index.
+ * @param {Array} list The array whose value at the supplied index will be replaced.
+ * @return {Array} A copy of the supplied array with the adjusted element at index `idx`.
+ * @example
+ *
+ *      R.adjust(R.add(10), 1, [0, 1, 2]);     //=> [0, 11, 2]
+ *      R.adjust(R.add(10))(1)([0, 1, 2]);     //=> [0, 11, 2]
+ */
+module.exports = _curry3(function(fn, idx, list) {
+  if (idx >= list.length || idx < -list.length) {
+    return list;
+  }
+  var start = idx < 0 ? list.length : 0;
+  var _idx = start + idx;
+  var _list = list.concat();
```

Comment:
If you want to support array-like objects, is there an existing function I should use for shallow cloning, or should I just use `Array.prototype.concat.call(list)` here?